providers.go

```diff
@@ -71,39 +70,66 @@ func file(s *Store, filename string, refresh bool, fn func([]byte) ([]Item, erro
 	inmem.Add(vals...)
 
 	if refresh {
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			errorProvider(s, providername, err)
+		}
+
 		go func() {
```

Comment:
Starting a goroutine without tracking and cancellation support. The `Store` should probably holds a context that would be passed to all goroutines that could then cancel their loop if the cancellation chan is closed.